plugins/com.google.cloud.tools.eclipse.googleapis/src/com/google/cloud/tools/eclipse/googleapis/internal/GoogleApiFactory.java

```diff
@@ -104,32 +97,31 @@ public void init() {
 
   @Override
   public Account getAccount() throws IOException {
-    return getAccount(GoogleCredential.getApplicationDefault());
+    return accountProvider.getAccount();
   }
-  Account getAccount(Credential credential) throws IOException {
-    HttpRequestInitializer chainedInitializer = new HttpRequestInitializer() {
-      @Override
-      public void initialize(HttpRequest httpRequest) throws IOException {
-        credential.initialize(httpRequest);
-        requestTimeoutSetter.initialize(httpRequest);
-      }
-    };
-    
-    Oauth2 oauth2 = new Oauth2.Builder(transport, jsonFactory, credential)
-        .setHttpRequestInitializer(chainedInitializer)
-        .setApplicationName(CloudToolsInfo.USER_AGENT)
-        .build();
-    
-    UserInfo userInfo = new UserInfo(oauth2.userinfo().get().execute());
-    return new Account(userInfo.getEmail(), credential, userInfo.getName(), userInfo.getPicture());
+  
+  @Override
+  public boolean hasCredentialsSet() throws IOException {
+    return accountProvider.hasCredentialsSet();
   }
   
+  private Credential getCredential() {
+    try {
+      return accountProvider.getCredential();
+    } catch (IOException ex) {
+      LOGGER.severe("Error when obtaining credential: " + ex);
```

Comment:
`+ ex` is effectively `+ ex.toString()` which will print the exception class name and maybe an instance ID. If you're at the point of logging an exception, always do it in a way that includes the stack trace.

```java
LOGGER.log(Level.SEVERE, "Error when obtaining credential.", ex);
```